README.md

```diff
@@ -1,103 +1,118 @@
 
-If you want to build an _über-jar_, execute the following command:
+## Prerequisites
 
-```shell script
-./mvnw package -Dquarkus.package.jar.type=uber-jar
-```
+- Java Development Kit (JDK) 17+
+- Maven 3.8.1+
+- Docker (optional, for container builds)
+- GraalVM (optional, for native compilation)
 
-The application, packaged as an _über-jar_, is now runnable using `java -jar target/*-runner.jar`.
+## Setup and Configuration
 
-## Creating a native executable
+### Local Development Setup
 
-You can create a native executable using:
+1. Clone the repository
+   ```bash
+   git clone <repository-url>
+   cd <repository-name>
+   ```
 
-```shell script
-./mvnw package -Dnative
+2. Install dependencies
+   ```bash
+   ./mvnw clean install
+   ```
+
+### Running the Application
+
+#### Development Mode
+```bash
+./mvnw compile quarkus:dev
 ```
+- Enables live coding
+- Access Dev UI at: http://localhost:8080/q/dev/
 
-Or, if you don't have GraalVM installed, you can run the native executable build in a container using:
+#### Production Build
+```bash
+# Regular JAR
+./mvnw package
+
+# Uber JAR
+./mvnw package -Dquarkus.package.jar.type=uber-jar
 
-```shell script
+# Native Executable
+./mvnw package -Dnative
+# Or with container build
 ./mvnw package -Dnative -Dquarkus.native.container-build=true
 ```
 
-You can then execute your native executable with: `./target/ptss-support-quarkus-1.0-SNAPSHOT-runner`
+### Branch Protection
+
+Detailed branch protection rules are configured for `development` and `main` branches. See the [Branch Protection](#setting-up-branch-protection-rules) section for specifics.
+
+### Code Quality
+
+#### SonarQube Integration
+1. Project is set up with SonarQube Cloud
+2. Automatic analysis is enabled
+3. New code definition can be customized
+
+## Documentation References
+
+### Internal Documentation
+- [Helm Configuration](/helm/README.md)
+- [CI/CD Documentation](/CI-CD.md)
+- [Authentication Filter](/src/main/kotlin/org/ptss/support/security/README.md)
+- [Code Formatting](/docs/code-formatting-spotless.md)
+- [SonarQube Local Setup](/docs/sonarqube-local.md)
 
-If you want to learn more about building native executables, please consult <https://quarkus.io/guides/maven-tooling>.
+### Related Repositories
+- [Platform Configuration](https://github.com/PTSS-Support/platform-config)
+- [Identity Service](https://github.com/PTSS-Support/Identity-Service)
+- [Questionnaire Service](https://github.com/PTSS-Support/Questionnaire-Service)
+- And more...
 
-## Related Guides
+## Technology Stack
 
-- Kotlin ([guide](https://quarkus.io/guides/kotlin)): Write your services in Kotlin
+- **Framework**: Quarkus
+- **Language**: Kotlin
+- **Build Tool**: Maven
+- **Code Quality**: SonarQube
+- **CI/CD**: GitHub Actions
 
-## Provided Code
+## Contributing
 
-### REST
+1. Create a feature branch from `development`
+2. Make your changes
+3. Ensure all tests pass
+4. Run SonarQube analysis
+5. Create a pull request to `development`
```

Comment:
Je maakt eerst een pr en dan wordt er automatisch gechecked of de tests passen en sonar analysis goed is